frontend2/src/views/Scrimmaging.tsx

```diff
@@ -0,0 +1,542 @@
+  }
+  return str;
+}
+
+const Scrimmaging: React.FC = () => {
+  const episodeId = useEpisodeId().episodeId;
+  const currentTeam = useCurrentTeam().team;
+
+  const [searchParams, setSearchParams] = useSearchParams();
+  const getParamEntries = (prev: URLSearchParams): Record<string, string> => {
+    return Object.fromEntries(prev.entries());
+  };
+
+  interface QueryParams {
+    inboxPage: number;
+    outboxPage: number;
+    teamsPage: number;
+    scrimsPage: number;
+    tourneyPage: number;
+    search: string;
+  }
+
+  const queryParams: QueryParams = useMemo(() => {
+    return {
+      inboxPage: parseInt(searchParams.get("inboxPage") ?? "1"),
+      outboxPage: parseInt(searchParams.get("outboxPage") ?? "1"),
+      teamsPage: parseInt(searchParams.get("teamsPage") ?? "1"),
+      scrimsPage: parseInt(searchParams.get("scrimsPage") ?? "1"),
+      tourneyPage: parseInt(searchParams.get("tourneyPage") ?? "1"),
+      search: searchParams.get("search") ?? "",
+    };
+  }, [searchParams]);
+
+  const [inboxLoading, setInboxLoading] = useState<boolean>(false);
+  const [outboxLoading, setOutboxLoading] = useState<boolean>(false);
+
+  const [searchText, setSearchText] = useState<string>(queryParams.search);
+  const [teamsLoading, setTeamsLoading] = useState<boolean>(false);
+  const [scrimsLoading, setScrimsLoading] = useState<boolean>(false);
+  const [tourneyLoading, setTourneyLoading] = useState<boolean>(false);
+
+  const [inboxData, setInboxData] = useState<
+    PaginatedScrimmageRequestList | undefined
+  >(undefined);
+  const [outboxData, setOutboxData] = useState<
+    PaginatedScrimmageRequestList | undefined
+  >(undefined);
+  const [teamsData, setTeamsData] = useState<
+    PaginatedTeamPublicList | undefined
+  >(undefined);
+  const [scrimsData, setScrimsData] = useState<PaginatedMatchList | undefined>(
+    undefined,
+  );
+  const [tourneyData, setTourneyData] = useState<
+    PaginatedMatchList | undefined
+  >(undefined);
+
+  function handleSearch(): void {
+    if (!teamsLoading && searchText !== queryParams.search) {
+      setSearchParams((prev) => ({
+        ...getParamEntries(prev),
+        teamsPage: "1",
+        search: searchText,
+      }));
+    }
+  }
+
+  useEffect(() => {
+    let isActiveLookup = true;
+    if (inboxLoading) return;
+    setInboxLoading(true);
+    setInboxData((prev) => ({ count: prev?.count ?? 0 }));
+
+    const fetchInbox = async (): Promise<void> => {
+      try {
+        const result = await getUserScrimmagesInbox(
+          episodeId,
+          queryParams.inboxPage,
+        );
+        if (isActiveLookup) {
+          setInboxData(result);
+        }
+      } catch (err) {
+        if (isActiveLookup) {
+          setInboxData(undefined);
+        }
+        console.error(err);
+      } finally {
+        setInboxLoading(false);
+      }
+    };
+
+    void fetchInbox();
+
+    return () => {
+      isActiveLookup = false;
+    };
+  }, [queryParams.inboxPage]);
```

Comment:
All of these useEffects also depend on episodeId, so we should add that into the dependencies